doc/languages-frameworks/gradle.section.md

```diff
@@ -0,0 +1,199 @@
+case, you will have to use the man-in-the-middle proxy.
+
+First, add a dependencies update script:
+
+```nix
+# this will use `pkgs.jextract` as the package to update deps for
+passthru.updateDeps = gradle.updateDeps { pname = "jextract"; };
+```
+
+Then run the update script via something like
+`$(nix-build -A <pname>.updateDeps)` (`nix-build` builds the
+`updateDeps` script, `$(...)` runs the script at the path printed by
+`nix-build`). It will generate `deps.json` in your package directory.
+You can then import it:
+
+```nix
+stdenv.mkDerivation {
+  # ...
+  mitmCache = gradle.fetchDeps {
+    inherit pname;
+    data = ./deps.json;
+  };
+
+  # required for using mitm-cache on Darwin
+  __darwinAllowLocalNetworking = true;
+  # ...
+}
+```
+
+If your package can't be evaluated using a simple `pkgs.<pname>`
+expression (for example, if your package isn't located in nixpkgs, or if
+you want to override some of its attributes), you must pass the `pkg`
+instead of `pname` to `gradle.updateDeps`. There are two ways of doing
+it.
+
+The first is to add the derivation arguments required for getting the
+package. Using the jextract example above:
+
+```nix
+{ lib
+, stdenv
+# ...
+, jextract
+}:
+
+stdenv.mkDerivation {
+  pname = "jextract";
+  # ...
+  passthru.updateDeps = gradle.updateDeps { pkg = jextract; };
+}
+```
+
+This allows you to `override` any arguments of the `pkg` used for
+`updateDeps` (for example, `pkg = jextract.override { enableSomeFlag =
+true };`), so this is the preferred way.
+
+The second is to create a `let` binding for the package, like this:
+
+```nix
+let self = stdenv.mkDerivation {
+  # ...
+  passthru.updateDeps = gradle.updateDeps { pkg = self; };
+}; in self
+```
+
+This is useful if you can't easily pass the derivation as its own
+argument, or if your `mkDerivation` is responsible for building multiple
+packages.
+
+In the former case, the update script will stay the same even if the
+derivation is called with different arguments. In the latter case, the
+update script will change depending on the derivation arguments. It's up
+to you to decide which one would work best for your derivation.
+
+## Update Script {#gradle-update-script}
+
+The update script flow:
+
+- Build the derivation's source via `pkgs.srcOnly`
+- Enter a `nix-shell` for the derivation in a `bwrap` sandbox (the
+  sandbox is only used on Linux)
+- Set the `IN_GRADLE_UPDATE_DEPS` environment variable to `1`
+- Run the derivation's `unpackPhase`, `patchPhase`, `configurePhase`
+- Run the derivation's `gradleUpdateScript` (the Gradle setup hook sets
+  a default value for it, which runs `preBuild`, `preGradleUpdate`
+  hooks, fetches the dependencies using `gradleUpdateTask`, and finally
+  runs the `postGradleUpdate` hook)
+- Finally, store all of the fetched files' hashes in the lockfile. They
+  may be `.jar`/`.pom` files from Maven repositories, or they may be
+  files otherwise used for building the package.
```

Comment:
It's unclear that this describes the internal operation of the update script; it reads like this is something users would have to do themselves.

Reply:
changed from "the update script flow" to "the update script does the following"